src/generator.cs

```diff
@@ -3393,37 +3399,364 @@ bool FilterMinimumVersion (AvailabilityBaseAttribute aa)
+			int revision = attr.Version.Revision == -1 ? 0 : attr.Version.Revision;
+			switch (attr.AvailabilityKind) {
+				case AvailabilityKind.Introduced:
+					return new IntroducedAttribute(platform, attr.Version.Major, attr.Version.Minor, revision, message: attr.Message);
+				case AvailabilityKind.Deprecated:
+					return new DeprecatedAttribute(platform, attr.Version.Major, attr.Version.Minor, revision, message: attr.Message);
+				case AvailabilityKind.Obsoleted:
+					return new ObsoletedAttribute(platform, attr.Version.Major, attr.Version.Minor, revision, message: attr.Message);
+				case AvailabilityKind.Unavailable:
+					return new UnavailableAttribute(platform, message: attr.Message);
+				default:
+					throw new NotImplementedException ();
+			}
+		}
+	}
+
+	static AvailabilityBaseAttribute CreateNoVersionSupportedAttribute (PlatformName platform)
+	{
+		switch (platform) {
+		case PlatformName.iOS:
+		case PlatformName.TvOS:
+		case PlatformName.MacOSX:
+			return new IntroducedAttribute(platform);
+		case PlatformName.WatchOS:
+			throw new InvalidOperationException ("CreateNoVersionSupportedAttribute for WatchOS never makes sense");
+		case PlatformName.MacCatalyst:
+			throw new InvalidOperationException ("CreateNoVersionSupportedAttribute for Catalyst never makes sense");
+		default:
+			throw new NotImplementedException ();
+		}
+	}
+
+	static AvailabilityBaseAttribute CreateUnsupportedAttribute (PlatformName platform)
+	{
+		switch (platform) {
+		case PlatformName.iOS:
+		case PlatformName.MacCatalyst:
+		case PlatformName.MacOSX:
+		case PlatformName.TvOS:
+			return new UnavailableAttribute(platform);
+		case PlatformName.WatchOS:
+			throw new InvalidOperationException ("CreateUnsupportedAttribute for WatchOS never makes sense");
+		default:
+			throw new NotImplementedException ();
+		}
+	}
+
+	HashSet<string> GetFrameworkListForPlatform (PlatformName platform)
+	{
+		HashSet<string> frameworkList = null;
+		switch (platform)
+		{
+			case PlatformName.iOS:
+				frameworkList = Frameworks.iosframeworks;
+				break;
+			case PlatformName.TvOS:
+				frameworkList = Frameworks.tvosframeworks;
+				break;
+			case PlatformName.MacOSX:
+				frameworkList = Frameworks.macosframeworks;
+				break;
+			case PlatformName.MacCatalyst:
+				frameworkList = Frameworks.maccatalystframeworks;
+				break;
+ 			default:
+				frameworkList = new HashSet<string>();
+				break;
+		}
+		return new HashSet<string>(frameworkList.Select(x => x.ToLower (CultureInfo.InvariantCulture)));
+	}
+
+	static string FindNamespace (MemberInfo item)
+	{
+		switch (item) {
+			case TypeInfo type:
+				return type.Namespace;
+			case PropertyInfo prop:
+				return prop.DeclaringType.Namespace;
+			case MethodInfo meth:
+				return meth.DeclaringType.Namespace;
+			default:
+				throw new NotImplementedException ($"FindNamespace on {item} of type {item.GetType()}");
+		}
+	}
```

Comment:
same here:
```
static string FindNamespace (MemberInfo item) => item switch {
    TypeInfo type => type.Namespace,
    PropertyInfo prop => prop.DeclaringType.Namespace,
    MethodInfo meth => meth.DeclaringType.Namespace,
   _ => throw new NotImplementedException ("..."),
}
```
Also - should this have a case for events?

Reply:
Events are never (?) in bindings, I'd rather crash and find out than add it and hope it works.